File: Ex12_5.m
%Ex12_5.m

clear out;

R=8.3143;    %J/mol-K

nc=3;          %성분수
m(1)=-1;        %stoichiometric number
a(1)=3.806;     %Cp_A
b(1)=1.566e-1;  %Cp_B
c(1)=-8.348e-5;  %Cp_C
d(1)=1.755e-8;  %Cp_D
HR(1)=52.51;    %기준온도에서의 del H0(KJ/mol)
GR(1)=68.43;    %기준온도에서의 del G0(KJ/mol)

m(2)=-1;        %stoichiometric number
a(2)=32.24;     %Cp_A
b(2)=1.924e-3;  %Cp_B
c(2)=1.055e-5;  %Cp_C
d(2)=-3.596e-9;  %Cp_D
HR(2)=-241.835;    %기준온도에서의 del H0(KJ/mol)
GR(2)=-228.614;    %기준온도에서의 del G0(KJ/mol)

m(3)=1;        %stoichiometric number
a(3)=9.014;     %Cp_A
b(3)=2.141e-1;  %Cp_B
c(3)=-8.39e-5;  %Cp_C
d(3)=1.373e-9;  %Cp_D
HR(3)=-234.95;    %기준온도에서의 del H0(KJ/mol)
GR(3)=-167.73;    %기준온도에서의 del G0(KJ/mol)

TR=298.15;      %기준온도(K)
T=145;          %반응온도(deg C)

for i=1:nc              % kJ/mol --> J/mol
    HR(i)=HR(i)*1000;
    GR(i)=GR(i)*1000;
end

T=T+273.15;

%delH0 and delG0 at T
[delH0, delG0]=DELHG_RXN(nc,m,a,b,c,d,TR,HR,GR,T);

%평형상수 K
K=exp(-delG0/R/T);

%print results
fprintf('del H0 at %8.3f deg C = %10.5f J/mol \n',T-273.15, delH0);
fprintf('del G0 at %8.3f deg C = %10.5f J/mol \n',T-273.15, delG0);
fprintf('평형상수 K at %8.3f deg C = %10.5f \n',T-273.15, K);

File: py/Ex12_5.py
import numpy as np
from DELHG_RXN import DELHG_RXN

R = 8.3143  # J/mol-K
nc = 3      # 성분수

# 각 성분별 데이터 배열 초기화
m = np.array([-1, -1, 1])
a = np.array([3.806, 32.24, 9.014])
b = np.array([1.566e-1, 1.924e-3, 2.141e-1])
c = np.array([-8.348e-5, 1.055e-5, -8.39e-5])
d = np.array([1.755e-8, -3.596e-9, 1.373e-9])
HR = np.array([52.51, -241.835, -234.95])    # kJ/mol
GR = np.array([68.43, -228.614, -167.73])    # kJ/mol

TR = 298.15     # 기준온도(K)
T_degC = 145    # 반응온도(deg C)

# kJ/mol을 J/mol로 변환
HR = HR * 1000.0
GR = GR * 1000.0

# 섭씨를 켈빈으로 변환
T = T_degC + 273.15

# delH0 and delG0 at T 계산 (외부 함수 호출)
delH0, delG0 = DELHG_RXN(nc, m, a, b, c, d, TR, HR, GR, T)

# 평형상수 K 계산[cite: 13]
K = np.exp(-delG0 / (R * T))

# 결과 출력[cite: 13]
print(f'del H0 at {T-273.15:8.3f} deg C = {delH0:10.5f} J/mol')
print(f'del G0 at {T-273.15:8.3f} deg C = {delG0:10.5f} J/mol')
print(f'평형상수 K at {T-273.15:8.3f} deg C = {K:10.5f}')